python/tests/builder/test_qalloc_init.py

```diff
@@ -0,0 +1,615 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+import pytest
+
+import cudaq
+import numpy as np
+
+skipIfNvidiaFP64NotInstalled = pytest.mark.skipif(
+    not (cudaq.num_available_gpus() > 0 and cudaq.has_target('nvidia-fp64')),
+    reason='Could not find nvidia-fp64 in installation')
+
+skipIfNvidiaNotInstalled = pytest.mark.skipif(
+    not (cudaq.num_available_gpus() > 0 and cudaq.has_target('nvidia')),
+    reason='Could not find nvidia in installation')
+
+# float
+
+
+@skipIfNvidiaFP64NotInstalled
+def test_kernel_float_params_f64():
+
+    cudaq.reset_target()
+    cudaq.set_target('nvidia-fp64')
```

Comment:
It's not for this PR, but at some point, we may need to update these new Python tests for state handling so that they can run on the `qpp` target, mainly to get them running as part of the PR CI pipeline (no GPU available).

Reply:
Sure!